### DISCUSSION TITLE
Error while trying to commit

### DISCUSSION THREAD

**Discussion by MBAatEJOT** [community/community#29012] [Codespaces]
Since a few weeks everytime I start my container, codespaces seems to add the following section to my `.git/config` file:

```
[commit]
	gpgsign = true
```

This leads to an error every time I'm trying to commit:

```
> git -c user.useConfigOnly=true commit --quiet --allow-empty-message --file -
error: gpg failed to sign the data
fatal: failed to write commit object
```

When I remove the section or set `gpgsign` to false the commit works. But when I restart the codespace it has reset to 

```
[commit]
	gpgsign = true
```

**Comment by joshaber** [comment]
Hi @MBAatEJOT, Codespaces should only be enabling GPG signing when it is enabled in your Codespaces settings at https://github.com/settings/codespaces:

<img width="794" alt="Screen Shot 2022-08-10 at 10 34 07 AM" src="https://user-images.githubusercontent.com/13760/183928814-5e5d56fb-69ef-4ef8-ab0d-b015fcffaac9.png">

So if you don't want GPG signing the easy solution is to disable that setting, and then new codespaces going forward should have it disabled. If you _do_ want GPG signing but it's consistently failing, let me know and we can try to dig into those failures more!

**Reply by MBAatEJOT** [reply]
Hmmm.... I don't remember activating this.... But that does not mean much.

Thanks @joshaber :-)
